src/Signer.js

```diff
@@ -3,30 +3,23 @@ import { MessageLayer } from 'streamr-client-protocol'
 const { StreamMessage } = MessageLayer
 const { SIGNATURE_TYPES } = StreamMessage
 
-import Web3 from 'web3'
-import FakeProvider from 'web3-fake-provider'
+import { ethers } from 'ethers'
 
 const debug = require('debug')('StreamrClient::Signer')
 
-const web3 = new Web3(new FakeProvider())
-
 export default class Signer {
     constructor(options = {}) {
         this.options = options
         if (this.options.privateKey) {
-            const account = web3.eth.accounts.privateKeyToAccount(this.options.privateKey)
-            this.address = account.address.toLowerCase()
-            this.sign = (d) => account.sign(d).signature
+            const wallet = new ethers.Wallet(this.options.privateKey)
+            this.address = wallet.address
+            this.sign = (d) => wallet.signMessage(d)
         } else if (this.options.provider) {
             this.sign = async (d) => {
-                const w3 = new Web3(this.options.provider)
-                const accounts = await w3.eth.getAccounts()
-                const address = accounts[0]
-                if (!address) {
-                    throw new Error('Cannot access account from provider')
-                }
-                this.address = address
-                return w3.eth.personal.sign(d, this.address)
+                const provider = new ethers.providers.Web3Provider(this.options.provider)
```

Comment:
Do we need to recreate everything inside the `sign` function? Maybe `provider` and `signer` can be created once and reused?

```
const provider = new ethers.providers.Web3Provider(this.options.provider)
const signer = provider.getSigner()
this.address = signer.address
this.sign = async (d) => {
                return signer.signMessage(d)
}
```

I'm not sure how this behaves if the active address changes (for example the user selects another account on Metamask). This could be tested to get a better understanding. Maybe that changes the `signer`, in which case we could still do:

```
const provider = new ethers.providers.Web3Provider(this.options.provider)
this.sign = async (d) => {
                const signer = provider.getSigner()
                this.address = signer.address
                return signer.signMessage(d)
}
```